### PR TITLE
[osh] adding PS1 parsing and evaluation

### PR DESCRIPTION
**Pull request by okayzed**
This is a continuation of #183 pull request, but this time from a branch in oilshell (instead of raisjn)

### CODE REVIEW

File: osh/word_parse.py [hunk 1]
```diff
@@ -269,6 +269,8 @@ def _ParseVarExpr(self, arg_lex_mode):
     if self.token_type == Id.Right_VarSub:
       return part  # no ops
 
+    log('%s', self.token_type)
```
Comment (okayzed) [inline osh/word_parse.py]: remove?

File: osh/lex.py [hunk 2]
```diff
@@ -445,6 +445,15 @@ def IsKeyword(name):
   R(r'[^\\\0]+', Id.Char_Literals),
 ]
 
+# https://www.gnu.org/software/bash/manual/html_node/Controlling-the-Prompt.html#Controlling-the-Prompt
+PS1_DEF = _C_STRING_COMMON + [
```
Comment (okayzed) [inline osh/lex.py]: is this okay? you mentioned refactoring multiple groups into one common group

File: core/word_eval.py [hunk 3]
```diff
@@ -552,7 +552,16 @@ def _EvalBracedVarSub(self, part, part_vals, quoted):
 
     elif part.suffix_op:
       op = part.suffix_op
-      if op.tag == suffix_op_e.StringUnary:
+      if op.tag == suffix_op_e.StringNullary:
+        if op.op_id == Id.VOp0_P:
+          # TODO: Use dependency injection
```
Comment (okayzed) [inline core/word_eval.py]: what's this comment?

File: osh/word_parse.py (hunk 1 above)
Comment (andychu) [reply osh/word_parse.py]: Hm not sure when that snuck in, I pushed a change to remove it.

File: osh/lex.py (hunk 2 above)
Comment (andychu) [reply osh/lex.py]: Discussed on Zulip, we should not use _C_STRING_COMMON

File: core/word_eval.py (hunk 3 above)
Comment (andychu) [reply core/word_eval.py]: This is to possibly replace the global ui.PROMPT with dependency injection.  We can leave it in for this PR, but I probably want to refactor it later to follow the prevailing style.

**Comment by andychu** [comment]
Merged!

**Pull request closed without merging**